src/math.rb

```diff
@@ -249,6 +249,20 @@ def log2(x)
       __call__('::log2', x)
     end
 
+    __function__('::log', ['double'], 'double')
+
+    def log(x)
+      raise TypeError, "can't convert String into Float" if x.is_a?(String)
+      begin
+        x = Float(x)
+      rescue ArgumentError
+        raise TypeError, "can't convert #{x.class.name} into Float"
+      end
+      return Float::NAN if x.nan?
+      raise DomainError, 'Numerical argument is out of domain' if x < 0
+      __call__('::log', x)
+    end
```

Comment:
How about adding a variadic argument?
(Plus the checks I forgot)
```suggestion
    def log(*x)
      if x.length == 1
        raise TypeError, "can't convert String into Float" if x.is_a?(String)
        begin
          x = Float(x)
        rescue ArgumentError
          raise TypeError, "can't convert #{x.class.name} into Float"
        end
        return Float::NAN if x.nan?
        raise DomainError, 'Numerical argument is out of domain' if x < 0
        __call__('::log', x)
      elsif x.length==2
        return log(x[0]) / log(x[1])
      else
        raise ArgumentError, 'Wrong number of arguments: #{x.length} (1 or 2 expected)'
      end
    end
```

Reply:
Implemented, thanks!